lib/open_project/journal_formatter/diff.rb

```diff
@@ -86,6 +80,29 @@ def link(key, options)
     end
   end
 
+  def url_attr(key, options)
+    journable = @journal.journable
+    version = @journal.version
+
+    if key.to_s == 'description'
+      default_attributes(options)
+      .merge(controller: '/journals',
+             action: 'diff',
+             id: @journal.id,
+             field: key.downcase,
+             activity_page: options[:activity_page])
+    else
+      default_attributes(options)
+      .merge(controller: '/wiki',
+             action: 'diff',
+             project_id: journable.project.identifier,
+             id: journable.page.slug,
+             version: version - 1,
+             version_from: version)
+    end
+    .compact
+  end
```

Comment:
I quite liked the third one as that would make the system more coherent and generic, but I understand you'd like to avoid a useless roundtrip to the server.

I do not like having too many different formatters. I thing extending `register_journal_formatted_fields` to make it possible to take options is a better approach.

Reply:
Well, it may be overkill to make it generic to support only one other use case. I would prefer option 1 as simplicity beats extensibility in this case in my opinion.

Reply:
@ulferts why do you not like the third option? It would keep the code simple and the path from journals to diff always the same, keeping the specifics of rendering a wiki diff in only one place.

Reply:
I have gone with the first option, using `OpenProject::JournalFormatter::WikiDiff` to deal with wiki diffs.

Reply:
@cbliard yes you might be right that it would also have been better than I thought initially. I wasn't so much thinking about the performance but rather about the indirections that I think are sometimes harder to follow. But the subclass will also do its job well.